fix: namespacesRequired undefined based on example

codenakama opened the pull request:
Adding `getInitialProps` call to fix namespacesRequired issue that occurs otherwise. Should be helpful to others as I made the same mistake.

isaachinman commented:
Hi there, can you explain what exactly this is fixing?

codenakama commented:
@isaachinman if you do not add the getInitialProps static method and return `pageProps`, then an error is thrown for `namespacesRequired` is undefined. I followed the example and found this issue, the fix is what I did on my PR. I just think it'd save time and hassle to other people next time they look for the example code.

isaachinman commented:
Hi @codenakama. If I clone this repo, and then do:

1. `yarn`
2. `yarn build`
3. `yarn build:examples/simple`
4. `yarn run-example`

I do not see any errors about `namespacesRequired`. Can you please provide steps to reproduce the problem you are describing?

codenakama commented:
@isaachinman indeed it works.

I also have tried going to simple example folder and:
1. `yarn`
2. `yarn dev`

throws error

```
arn run v1.15.2
$ node index.js
/Users/ricardoabreu/Development/open-source/next-i18next/dist/commonjs/components/Link.js:130
var _default = (0, _reactI18next.withTranslation)()(Link);
                                                 ^

TypeError: (0 , _reactI18next.withTranslation) is not a function
    at Object.<anonymous> (/Users/ricardoabreu/Development/open-source/next-i18next/dist/commonjs/components/Link.js:130:50)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/ricardoabreu/Development/open-source/next-i18next/dist/commonjs/components/index.js:21:36)
    at Module._compile (module.js:653:30)
error Command failed with exit code 1.
```

But I don't know what's the difference... I created a simple next app, followed all the steps and got the error as I mentioned in my original post, I added the getInitialProps and the issue went away.
I just tried again to validate, removed the `getInitialProps` as per my fix in this PR and I get 
```
Cannot read property 'namespacesRequired' of null
```

isaachinman commented:
The simple example requires linking to work properly at runtime.

Please provide a reproducible example of the issue you are describing.

codenakama commented:
I can't replicate the issue on a separate project, so I'd assume it's a problem with some other package perhaps, however I started getting different issues where I could not get /en, /fr to work and was getting 404 page errors although I did setup everything based on the docs. Im happy for this PR to be abandoned, I don't have more time to deal with this at the moment and I also don't want to waste more of yours.

isaachinman commented:
No problem. If you figure out how to reproduce the problem please do just let me know and I'll debug it.
